mint-client/src/api.rs

```diff
@@ -48,6 +49,30 @@ impl<'a> dyn FederationApi + 'a {
             }
         }
     }
+    pub async fn await_output_outcome<T: TryIntoOutcome + Send>(
+        &self,
+        outpoint: OutPoint,
+        timeout: Duration,
+    ) -> Result<T> {
+        let poll = || async {
+            let interval = Duration::from_secs(1);
+            loop {
+                match self.fetch_output_outcome(outpoint).await {
+                    Ok(t) => return Ok(t),
+                    Err(e) => {
+                        if e.is_retryable_fetch_coins() {
+                            tokio::time::sleep(interval).await;
+                        } else {
+                            return Err(e);
+                        }
+                    }
```

Comment:
not bad here, but in case you aren't aware: you can [guard match arms](https://doc.rust-lang.org/rust-by-example/flow_control/match/guard.html)

Reply:
Fixed this